### PR TITLE
Xlgpp unit test

### PR DESCRIPTION
**Pull request by sahilmgandhi**
[AB#1574150](https://dev.azure.com/mseng/708e929f-6bd5-415a-8daf-25b1dac08dd8/_workitems/edit/1574150) 

A basic unit test for the new xlg to db analyzer that checks that a new DB was created and the files (.sst) have been created. 
Invoke using `bxl Test.Tool.Analyzers.dsc`

### CODE REVIEW

File: Public/Src/Tools/UnitTests/Analyzers/XLGPlusPlusAnalyzersTests.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        }
+
+        /// <summary>
+        /// TODO: Once all events can be successfully stored, make sure that the count in xlg = count in DB
+        /// </summary>
+        [Fact]
+        public void ValidateEventCount()
+        {
+            XAssert.AreEqual(5, 5);
+        }
```
Comment (olkononenko) [inline Public/Src/Tools/UnitTests/Analyzers/XLGPlusPlusAnalyzersTests.cs]: i'd nuke it from this PR. We do not generally keep placeholders in our code

File: Public/Src/Tools/UnitTests/Analyzers/XLGPlusPlusAnalyzersTests.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+using Test.BuildXL.Scheduler;
+using Test.BuildXL.TestUtilities.Xunit;
+using BuildXL.Execution.Analyzer;
+using Xunit;
+using Xunit.Abstractions;
+using static BuildXL.ToolSupport.CommandLineUtilities;
+using static Test.Tool.Analyzers.AnalyzerTestBase;
+using BuildXLConfiguration = BuildXL.Utilities.Configuration;
+using System;
+using System.Diagnostics;
```
Comment (olkononenko) [inline Public/Src/Tools/UnitTests/Analyzers/XLGPlusPlusAnalyzersTests.cs]: sort usings and nuke the unused ones (there is command in VS that will do it for you)

File: Public/Src/Tools/UnitTests/Analyzers/XLGPlusPlusAnalyzersTests.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+{
+    /// <summary>
+    /// Tests for <see cref="XLGToDBAnalyzer"/>
+    /// The construction and disposal of these tests rely on the fact that 
+    /// Xunit uses a unique class instance for each test (as is with the other Analyzer Tests).
+    /// </summary>
+    public class XLGPlusPlusAnalyzersTests : AnalyzerTestBase
+    {
+
+        private AbsolutePath DirPath { get; set; }
```
Comment (olkononenko) [inline Public/Src/Tools/UnitTests/Analyzers/XLGPlusPlusAnalyzersTests.cs]: nit : empty line

File: Public/Src/Tools/UnitTests/Analyzers/XLGPlusPlusAnalyzersTests.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+using System.Diagnostics;
+
+namespace Test.Tool.Analyzers
+{
+    /// <summary>
+    /// Tests for <see cref="XLGToDBAnalyzer"/>
+    /// The construction and disposal of these tests rely on the fact that 
+    /// Xunit uses a unique class instance for each test (as is with the other Analyzer Tests).
+    /// </summary>
+    public class XLGPlusPlusAnalyzersTests : AnalyzerTestBase
```
Comment (olkononenko) [inline Public/Src/Tools/UnitTests/Analyzers/XLGPlusPlusAnalyzersTests.cs]: >XLGPlusPlusAnalyzersTests [](start = 17, length = 25)

the name does not align with `XLGToDBAnalyzer`

File: Public/Src/Tools/UnitTests/Analyzers/XLGPlusPlusAnalyzersTests.cs [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        /// This test makes sure that a RocksDB database was created and populated with some sst files.
+        /// </summary>
+        [Fact]
+        public void DBCreatedAndPopulated()
+        {
+            Configuration.Logging.LogExecution = true;
+
+            var file = FileArtifact.CreateOutputFile(Combine(DirPath, "blah.txt"));
+
+            var pipB = CreateAndSchedulePipBuilder(new Operation[]
```
Comment (olkononenko) [inline Public/Src/Tools/UnitTests/Analyzers/XLGPlusPlusAnalyzersTests.cs]: >B [](start = 19, length = 1)

where is pipA? :)

File: Public/Src/Tools/UnitTests/Analyzers/XLGPlusPlusAnalyzersTests.cs [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+            {
+                Operation.ReadFile(file),
+                Operation.WriteFile(CreateOutputFileArtifact())
+            }).Process;
+
+            var buildA = RunScheduler().AssertSuccess();
+
+            var analyzerRes = RunAnalyzer(buildA).AssertSuccess();
+
+            XAssert.AreNotEqual(Directory.GetFiles(DirPath.ToString(Context.PathTable), "*.sst"), 0);
```
Comment (olkononenko) [inline Public/Src/Tools/UnitTests/Analyzers/XLGPlusPlusAnalyzersTests.cs]: >Directory.GetFiles(DirPath.ToString(Context.PathTable), "*.sst"), 0 [](start = 32, length = 67)

it's not clear right away what exactly you are asserting here. Add a call to Length property

Comment (olkononenko) [inline Public/Src/Tools/UnitTests/Analyzers/XLGPlusPlusAnalyzersTests.cs]: >XAssert.A [](start = 12, length = 9)

add a small comment explaining the meaning of "there should be .sst files"

File: Public/Src/Tools/UnitTests/Analyzers/XLGPlusPlusAnalyzersTests.cs (hunk 2 above)
Comment (sahilmgandhi) [reply Public/Src/Tools/UnitTests/Analyzers/XLGPlusPlusAnalyzersTests.cs]: CtrlR + CtrlG

---
In reply to: [306919228](https://github.com/microsoft/BuildXL/pull/650#discussion_r306919228) [](ancestors = 306919228)

File: Public/Src/Tools/UnitTests/Analyzers/XLGPlusPlusAnalyzersTests.cs (hunk 5 above)
Comment (sahilmgandhi) [reply Public/Src/Tools/UnitTests/Analyzers/XLGPlusPlusAnalyzersTests.cs]: Whoops forgot to rename to A and B. 

---
In reply to: [306923214](https://github.com/microsoft/BuildXL/pull/650#discussion_r306923214) [](ancestors = 306923214)

**Pull request merged**